guides/micronaut-java-records/java/src/main/java/example/micronaut/BookController.java

```diff
@@ -0,0 +1,43 @@
+package example.micronaut;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.http.annotation.Controller;
+import io.micronaut.http.annotation.Get;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Controller("/books") // <1>
+class BookController {
+
```

Comment:
As we are accessing a db from the controller we should annotate it with `@ExecuteOn(...)`. Either at class level or method level

Reply:
Good catch